voir/instruments/gpu/__init__.py

```diff
@@ -0,0 +1,152 @@
+import os
+import time
+import glob
+import traceback
+from threading import Thread
+
+from ...tools import instrument_definition
+from ..utils import Monitor as Monitor2
+
+
+def find_monitors():
+    """Look for device monitor implementation (AMD or ROCm)"""
+    backends = {}
+    base = __file__
+    base_module = "voir.instruments.gpu"
+    module_path = os.path.dirname(os.path.abspath(base))
+    pattern = os.path.join(module_path, "[A-Za-z]*")
+
+    for module_path in glob.glob(pattern, recursive=False):
+        module_file = module_path.split(os.sep)[-1]
+
+        if module_file == "__init__.py":
+            continue
+
+        module_name = module_file.split(".py")[0]
+
+        try:
+            module = __import__(".".join([base_module, module_name]), fromlist=[""])
+        except ImportError:
+            print(traceback.format_exc())
+            continue
+
+        backends[module_name] = module
+
+    return backends
+
+
+BACKENDS = find_monitors()
+BACKEND = None
+MONITOR = None
+ARCH = None
```

Comment:
Do we need these globals?

Reply:
yeah, it is to avoid creating multiple monitor/caching the selected arch